symengine/parser.cpp

```diff
@@ -0,0 +1,242 @@
+#include <symengine/basic.h>
+#include <symengine/symbol.h>
+#include <symengine/add.h>
+#include <symengine/integer.h>
+#include <symengine/rational.h>
+#include <symengine/complex.h>
+#include <symengine/mul.h>
+#include <symengine/pow.h>
+#include <symengine/functions.h>
+#include <symengine/constants.h>
+#include <symengine/visitor.h>
+#include <vector>
+#include <stack>
+#include <map>
+#include <set>
+#include <string>
+#include <functional>
+namespace SymEngine {
+
+class ExpressionParser
+{
+    std::set<char> OPERATORS = {'-', '+', '*', '/', '^', '(', ')', ','};
+    std::map<char, int> op_precedence = {
+        {')', 0}, {',', 0}, {'-', 1},  {'+', 1},
+        {'*', 3}, {'/', 4}, {'^', 5}
+    };
+    std::map<std::string, RCP<const Basic> > constants = {
+        {"e", E}, {"EulerGamma", EulerGamma}, {"pi", pi}, {"i", I}
```

Comment:
I'll do the changes, but I feel 'e' should be used for natural logarithm, as it is the universal symbol. (I'm neutral about the rest)